File: appjava/src/main/java/io/firekast/appjava/MainActivity.java
package io.firekast.appjava;

import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

public class MainActivity extends AppCompatActivity {

    private BottomNavigationView mBottomNavigationView;

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener = new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            return navigateTo(item.getItemId());
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mBottomNavigationView = findViewById(R.id.navigation);
        mBottomNavigationView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);

        navigateTo(mBottomNavigationView.getSelectedItemId());
    }

    private boolean navigateTo(@IdRes int navigationItemId) {
        if (mBottomNavigationView.getSelectedItemId() == navigationItemId
                && !getSupportFragmentManager().getFragments().isEmpty()) {
            return false;
        }
        switch (navigationItemId) {
            case R.id.navigation_streamer:
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.container_main, new StreamerFragment())
                        .commit();
                return true;
            case R.id.navigation_player:
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.container_main, new PlayerFragment())
                        .commit();
                return true;
        }
        return false;
    }

}

File: app/src/main/java/io/firekast/demo/MainActivity.kt
package io.firekast.demo

import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        if (navigation.selectedItemId == item.itemId && !supportFragmentManager.fragments.isEmpty()) {
            return@OnNavigationItemSelectedListener false
        }
        when (item.itemId) {
            R.id.navigation_streamer -> {
                supportFragmentManager.beginTransaction()
                        .replace(R.id.fragmentContainer, StreamerFragment())
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_player -> {
                supportFragmentManager.beginTransaction()
                        .replace(R.id.fragmentContainer, Fragment())
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
        navigation.selectedItemId = R.id.navigation_streamer
    }
}
